### ISSUE TITLE
Construction of networks and calculation of network metrics

### ISSUE THREAD

**Issue by jonrhodes** [sLandserve/sLandserv#1]
Suggest considering the following:

1) In "create_ee_network.R" calculate edge to edge distances rather than centroid to centroid distances on lines 40-41 as I think this makes better ecological sense. I think the same should also be considered for the distances in "create_es_network.R"

2) On line 50 in "create_ee_network.R", where we create the igraph network, we need "diag=TRUE" as we are permitting self loops in the calculation of connectivity effects for supply in the simulation. The network should be consistent with this. 

3) On lines 51-55 in "create_ee_network.R", where we calculate the network metrics, we need to specify "loops=TRUE"  in the use of edge_density(), centr_degree() and degree() functions to reflect the fact that we are permitting self loops in the calculation of connectivity effects for supply. Again, the network calculations need to consistent with this assumption. Note that, in the supply-demand network, there are no self loops assumed, but some of the network metric functions such as centr_degree() and degree() set loops=TRUE by default - need to check - possibly it doesn't matter as this is a bipartite network.

4) In the "calculate_benefit.R" function it would be very useful for interpretation if it could also output the total supply. This would help disentangle the different effects of landscape structure on supply and benefit.

**Comment by laurajanegraham** [comment]
Hi Jonathan, all of these sound sensible. I'll aim to implement them this week or next.

**Comment by jonrhodes** [comment]
Thanks Laura. Happy to help where I can. Maybe once we have addressed these issues we could get another set of simulations run?

**Comment by jonrhodes** [comment]
I have been looking at the code for generating the networks and, although create_ee_network() and create_es_network() do use centroid to centroid distances, it looks like network_create() uses edge to edge, as far as I can see, which is what we want. Since we are using network_create() for the simulations it seems like there isn't an issue here wrt the distance calculations and maybe we just don't need create_ee_network() and create_es_network() anymore?

**Comment by jonrhodes** [comment]
Ahh, I see now that you do use create_ee_network() and create_es_network(). I have fixed network_create to deal with the way the new ls_create() now generates supply and demand, the issue of edge to edge distances, and self-loops in the network metrics. but haven't touched create_ee_network() and create_es_network(). All my changes are in the separate fork.

**Comment by laurajanegraham** [comment]
Looks like we've both been working on this at the same time. The master branch now has all the required changes (including the updated landscape simulation, which I've checked through and amended where needed). I've also got rid of unused files (for example network_create.R, which has been replaced by create_ee_network() and create_es_network(). 

I'll set a new lot of simulations running soon, but they'll be behind some other stuff in the queue on the HPC (so hopefully will have them completed before the end of the week). 

We can delete the test_new_landscapes branch.
